sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -1339,6 +1340,31 @@ def test_issue_14787():
     f = Piecewise((x, x < 1), ((S(58) / 7), True))
     assert str(f.evalf()) == "Piecewise((x, x < 1), (8.28571428571429, True))"
 
+def test_issue_21481():
+    b, e = symbols('b e')
+    A = Piecewise((1, Eq(b, 1) | Eq(e, 0) | (Eq(b, -1) & Eq(Mod(e, 2), 0))),
+                    (0, Eq(b, 0) & (e > 0)), (-1, Eq(b, -1) & Eq(Mod(e, 2), 1)),
+                    (Piecewise((2, ((b > 1) & (e > 0)) | ((b > 0) & (b < 1) & (e < 0)) |
+                            ((e >= 2) & (b < -1) & Eq(Mod(e, 2), 0)) |
+                            ((e <= -2) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 0))),
+                            (S.Half, ((b > 1) & (e < 0)) | ((b > 0) & (e > 0) & (b < 1)) |
+                            ((e <= -2) & (b < -1) & Eq(Mod(e, 2), 0)) |
+                            ((e >= 2) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 0))),
+                            (-S.Half, Eq(Mod(e, 2), 1) & (((e <= -1) & (b < -1)) |
+                            ((e >= 1) & (b > -1) & (b < 0)))),
+                                (-2, ((e >= 1) & (b < -1) & Eq(Mod(e, 2), 1)) |
+                                ((e <= -1) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 1)))),
+                    Eq(im(b), 0) & Eq(im(e), 0)))
+    B = piecewise_fold(A)
+    sa = A.simplify()
+    sb = B.simplify()
+    v = Tuple(-2, -1, -0.5, 0, 0.5, 1, 2)
```

Comment:
Why do you have to use `Tuple` instead of tuple? And is it intended to use `0.5` instead of `S.Half`?

Reply:
Not a specific idea, but Tuple would be better I thought.

Reply:
should I replace it ?

Reply:
And also I suggest to use `S.Half` instead of floats.